ledger/babbage.go

```diff
@@ -209,6 +209,9 @@ func (b *BabbageTransactionBody) ReferenceInputs() []TransactionInput {
 }
 
 func (b *BabbageTransactionBody) CollateralReturn() TransactionOutput {
+	if b.TxCollateralReturn == nil {
+		return nil
+	}
```

Comment:
This seems redundant, because you would return nil anyway

Reply:
I added a comment in the code about why we're doing that